### PR TITLE
Enable OAuth2 refresh token.

### PR DESCRIPTION
**Pull request by mshima**
- Implement OAuth2{Reactive}RefreshTokensWebFilter.java and instantiate dependencies.
- Inject OAuth token/user into required services for tests.
- Fix angular requiring api/account twice, causing problem with user synchronization.

Fixes https://github.com/jhipster/generator-jhipster/issues/15069.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/resources/logback-spring.xml.ejs [hunk 1]
```diff
@@ -45,6 +45,7 @@
     </root>
 -->
 
+    <logger name="<%= packageName %>" level="INFO"/>
```
Comment (DanielFran) [inline generators/server/templates/src/main/resources/logback-spring.xml.ejs:48]: This is not needed, already defined in application.yml

Comment (mshima) [reply generators/server/templates/src/main/resources/logback-spring.xml.ejs:48]:
```suggestion
```

**Comment by Tcharl** [comment]
Right, using filter is way better than what we did.

File: generators/server/files.js [hunk 2]
```diff
@@ -1165,6 +1165,35 @@ const serverFiles = {
       templates: ['META-INF/services/reactor.blockhound.integration.BlockHoundIntegration'],
     },
   ],
+  springBootOauth2: [
+    {
+      condition: generator => generator.authenticationType === 'oauth2' && generator.applicationType === 'monolith',
```
Comment (Tcharl) [inline generators/server/files.js:1170]: constants?

File: generators/server/templates/src/main/java/package/service/UserService.java.ejs [hunk 3]
```diff
@@ -999,7 +999,16 @@ public class UserService {
             })
             <%_ } _%>
             .collect(Collectors.toSet()));
-        return <% if (databaseType !== 'no' && !reactive) { %>new <%= asDto('AdminUser') %>(syncUserWithIdP(attributes, user))<% } else if (!reactive) { %>user<% } %><% if (databaseType === 'no' && reactive) { %>Mono.just(user)<% } else if (reactive) { %>syncUserWithIdP(attributes, user).flatMap(u -> Mono.just(new <%= asDto('AdminUser') %>(u)))<% } %>;
+
+    <%_ if (databaseType === 'no') { _%>
```
Comment (Tcharl) [inline generators/server/templates/src/main/java/package/service/UserService.java.ejs:1003]: constants?

**Comment by Tcharl** [comment]
Everything looks good now: do you want I merge it?

**Comment by mshima** [comment]
> Everything looks good now: do you want I merge it?

If it looks ok, then yes 😄.

File: generators/server/templates/src/test/java/package/web/rest/LogoutResourceIT.java.ejs [hunk 4]
```diff
@@ -66,23 +65,30 @@ class LogoutResourceIT {
     @Autowired
     private <% if (!reactive) { %>Web<% } %>ApplicationContext context;
 
+    @Autowired
+    private <%= reactivePrefix %>OAuth2AuthorizedClientService authorizedClientService;
+
+    @Autowired
+	private ClientRegistration clientRegistration;
```
Comment (mraible) [inline generators/server/templates/src/test/java/package/web/rest/LogoutResourceIT.java.ejs:72]: Indent 4 spaces

**Review by mraible — COMMENTED** [review]
We should add `offline_access` as a scope and configure Keycloak to return a refresh token by default.

**Review by mraible — APPROVED** [review]
😃🤩✌️

**Comment by mshima** [comment]
> We should add `offline_access` as a scope and configure Keycloak to return a refresh token by default.

@mraible in my tests `offline_access` is not required to enable refresh tokens, it's already enabled by default.

At least our angular client, uses cookies, which expires and I think this makes `offline_access` useless.
And we don't provide a mobile app, so IMO `offline_access` is not a sensible default.

**Comment by mraible** [comment]
We have React Native, Ionic, and Flutter support! 😊

I think we should have it enabled by default in dev mode, maybe not in production.

**Comment by Tcharl** [comment]
@mraible offline access is achieved at Keycloak level, not spring-sec one right? Also, the claim is already configured on  keycloak side.
Merging it, from what I've seen & tested everything is ok and much better than before

**Comment by mraible** [comment]
The client (Spring Security in this case) needs to pass in `offline_access` as a scope in order to get a refresh token. If one is returned without this scope, that seems strange.

**Pull request merged**